business/apps.go

```diff
@@ -57,6 +57,14 @@ func buildFinalLabels(m map[string][]string) map[string]string {
 
 // GetAppList is the API handler to fetch the list of applications in a given namespace
 func (in *appService) GetAppList(ctx context.Context, namespace string, linkIstioResources bool) (models.AppList, error) {
+	var end observability.EndFunc
+	ctx, end = observability.StartSpan(ctx, "GetAppList",
```

Comment:
+1 on this approach.
A couple of suggestions, I don't think that the parameters would need to be strong-typed, perhaps they could be moved to strings all the time (are just used to fill the tags and log).
Also, I think that all business methods should call the "StartSpan", no need to have two implementations in the business/layer, the observability package can handle the "enabled/disabled" I think. The cost would be to have an empty call, which wouldn't cause any performance, but it would reduce the boilerplate of the business/layer side.